hudi-utilities/src/main/java/org/apache/hudi/utilities/HoodieCompactor.java

```diff
@@ -271,9 +262,9 @@ private int doCompact(JavaSparkContext jsc) throws Exception {
     }
   }
 
-  private Option<String> doSchedule(JavaSparkContext jsc) {
+  private Option<String> doSchedule(JavaSparkContext jsc) throws Exception {
     try (SparkRDDWriteClient client =
-             UtilHelpers.createHoodieClient(jsc, cfg.basePath, "", cfg.parallelism, Option.of(cfg.strategyClassName), props)) {
+             UtilHelpers.createHoodieClient(jsc, cfg.basePath, getSchema(), cfg.parallelism, Option.of(cfg.strategyClassName), props)) {
```

Comment:
Does this fail with empty schema string?  If so, could you construct a test case that fails without the fix and passes with the fix?

Reply:
Will do.

Reply:
@jonvex, do you know how to trigger the failures by setting "hoodie.schema.add.field.ids = true"?

Reply:
As discussed offline, a test implementation of `HoodieAvroWriteSupport` needs to be added, that does schema parsing to reproduce the failure.  This is related to the custom write support as a Hudi extension in XTable: https://github.com/apache/incubator-xtable/blob/main/hudi-support/extensions/README.md.